packages/tasit-account/src/ethers.test.js

```diff
@@ -1,42 +1,19 @@
 import { expect, assert } from "chai";
 import { ethers } from "ethers";
 
-const waitForEvent = async (eventName, expected) => {
-  return new Promise(function(resolve, reject) {
-    contract.on(eventName, function() {
-      const args = Array.prototype.slice.call(arguments);
-      const event = args.pop();
-      event.removeListener();
-      expect(
-        args,
-        `${event.event} event should have expected args`
-      ).to.deep.equal(expected);
-      resolve();
-    });
-  });
-};
-
 const zero = ethers.utils.bigNumberify(0);
 
-let wallet, randomWallet, contract;
-
-const rawTx = {
-  nonce: 0,
-  gasLimit: 21000,
-  gasPrice: ethers.utils.bigNumberify("20000000000"),
-  to: "0x88a5C2d9919e46F883EB62F7b8Dd9d0CC45bc290",
-  value: ethers.utils.parseEther("0.0000001"),
-  data: "0x",
-};
-
-// Note: Once we refactor this test suite for each test to require no prior state
-// other than what is in that describe's beforeEach hook, we'll give a more
-// descriptive name to this describe block
-describe("ethers.js - pretests", function() {
-  const provider = new ethers.providers.JsonRpcProvider();
-  provider.pollingInterval = 50;
-
-  it("should create a wallet from priv key and provider", async function() {
+let wallet, provider;
```

Comment:
These vars are assigned beforeEach test case, because that `not defined` errors will occur if I'll do that. Do you know a better solution?

Reply:
If you define the const for the first time within each test case, I think that would work.

Reply:
Sorry but I'm really not understood that.
```
  it("should get contract's value", async () => {
    //const contract; SyntaxError: Unexpected token (should be assigned when declared)
    //const contract = undefined; TypeError: Cannot read property 'getValue' of undefined
    const value = await contract.getValue();
    expect(value).to.exist;
    expect(value).to.be.a("string");
  });
```
Sample from mocha doc (using var): https://mochajs.org/#asynchronous-hooks

Reply:
Yeah, I think we want to assign things when declaring them wherever possible.

This comment is about wallet and provider as opposed to contract, so let's use one of the wallet tests for an example.

What you're saying is we want to use `wallet` that's assigned in the beforeEach in the test "should get balance of wallet". And it wouldn't work to declare it in the `beforeEach` because then it would be undefined in the `it`. Is that right?

https://github.com/marcelomorgado/TasitSDK/blob/25c82719d262ed3dd37db55d99a979e67373ffd1/packages/tasit-account/src/ethers.test.js#L30

If so, that makes sense and no change is needed here. I was just making sure both wallet and provider were (1) used in multiple tests and (2) used in the `beforeEach`. I think that's the only time we want to declare a variable this high up, right?

Reply:
> What you're saying is we want to use wallet that's assigned in the beforeEach in the test "should get balance of wallet". And it wouldn't work to declare it in the beforeEach because then it would be undefined in the it. Is that right?

Yes.

> If so, that makes sense and no change is needed here. I was just making sure both wallet and provider were (1) used in multiple tests and (2) used in the beforeEach. I think that's the only time we want to declare a variable this high up, right?

Yes, I'm trying to always use `const` inside test cases whenever possible.